core/src/main/scala/io/dantb/contentless/ContentModel.scala

```diff
@@ -85,20 +95,60 @@ object Validation:
+          val (allButOne, last) = allowedValues.map(_.asString).splitAt(size - 1)
+          s"Only ${allButOne.mkString(", ")} and ${last.head} marks are allowed".some
+
+  final case class RichTextNodeTypes(allowedValues: Set[RichTextNodeType]) extends Validation:
+    def message: Option[String] =
+      val size = allowedValues.size
+      size match
+        case 0 => None
+        case 1 => s"Only ${allowedValues.head.asString} nodes are allowed".some
+        case other =>
+          val (allButOne, last) = allowedValues.map(_.asString).splitAt(size - 1)
+          s"Only ${allButOne.map(_.replaceAll("-", " ")).mkString(", ")} and ${last.head} nodes are allowed".some
+
+  final case class Size(min: Option[Int], max: Option[Int], message: Option[String]) extends Validation
+  final case class DateRange(min: Option[ZonedDateTime], max: Option[ZonedDateTime]) extends Validation
+  case object Unique                                                                 extends Validation
+
+  enum Regexp(val underlying: Regex) extends Validation:
+    case Url
+        extends Regexp("^(ftp|http|https):\\/\\/(\\w+:{0,1}\\w*@)?(\\S+)(:[0-9]+)?(\\/|\\/([\\w#!:.?+=&%@!\\-\\/]))?$".r)
+    case Email                  extends Regexp("^\\w[\\w.-]*@([\\w-]+\\.)+[\\w-]+$".r)
+    case Slug                   extends Regexp("^[a-z0-9]+(?:-[a-z0-9]+)*$".r)
+    case DateUS                 extends Regexp("^(0?[1-9]|1[012])[- /.](0?[1-9]|[12][0-9]|3[01])[- /.](19|20)?\\d\\d$".r)
+    case DateEurope             extends Regexp("^(0?[1-9]|[12][0-9]|3[01])[- /.](0?[1-9]|1[012])[- /.](19|20)?\\d\\d$".r)
+    case Time12H                extends Regexp("^(0?[1-9]|1[012]):[0-5][0-9](:[0-5][0-9])?\\s*[aApP][mM]$".r)
+    case Time24H                extends Regexp("^(0?[0-9]|1[0-9]|2[0-3]):[0-5][0-9](:[0-5][0-9])?$".r)
+    case PhoneNumUS             extends Regexp("^\\d[ -.]?\\(?\\d\\d\\d\\)?[ -.]?\\d\\d\\d[ -.]?\\d\\d\\d\\d$".r)
+    case ZipCodeUS              extends Regexp("^\\d{5}$|^\\d{5}-\\d{4}$}".r)
+    case Custom(pattern: Regex) extends Regexp(pattern)
+
+  object Regexp:
+    def unapply(r: Regexp): Some[Regex] = Some(r.underlying)
+
+    given Eq[Regexp] = Eq.fromUniversalEquals
+
+    def of(raw: String): Regexp =
+      if raw == Regexp.Url.underlying.toString() then Regexp.Url
+      else if raw == Regexp.Email.underlying.toString() then Regexp.Email
+      else if raw == Regexp.Slug.underlying.toString() then Regexp.Slug
+      else if raw == Regexp.DateUS.underlying.toString() then Regexp.DateUS
+      else if raw == Regexp.DateEurope.underlying.toString() then Regexp.DateEurope
+      else if raw == Regexp.Time12H.underlying.toString() then Regexp.Time12H
+      else if raw == Regexp.Time24H.underlying.toString() then Regexp.Time24H
+      else if raw == Regexp.PhoneNumUS.underlying.toString() then Regexp.PhoneNumUS
+      else if raw == Regexp.ZipCodeUS.underlying.toString() then Regexp.ZipCodeUS
+      else Regexp.Custom(raw.r)
```

Comment:
Not saying it's any better but you could do `List(Regexp.Url, Regexp.Email, ...).find(r => raw == r.underlying.toString()).getOrElse(Regexp.Custom(raw.r)`